### PR TITLE
Terminology: suite to workflow

### PR DESCRIPTION
**Pull request by hjoliver**
This is a ~~small~~ *large-but-trivial* change with no associated Issue (but discussed with the team!)

- globally replace suite/Suite/SUITE with workflow/Workflow/WORKFLOW
  - thank @@#$$@# for easy command-line automation
- rename all `*suite*` filenames as `*workflow*`
- restore any back-compat references to `suite.rc`
- fix style violations resulting from slightly longer lines
- restore `CYLC_SUITE_*` job environment variables for back-compat

Note:
- much easier to do this all at once than separate user-facing and internal changes
- some use of the word *suite* should really be translated to *scheduler* rather than *workflow*, but I haven't done that (requires a careful look at each case)

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Already covered by existing tests.
- [x] Appropriate change log entry included.
- [x] documentation PR: cylc/cylc-doc/issues/238
- [x] No dependency changes.

### CODE REVIEW

**Comment by hjoliver** [comment]
All tests passing locally.

File: cylc/flow/parsec/fileparse.py
Comment (datamel) [inline cylc/flow/parsec/fileparse.py:218]:
```suggestion
    # Load Rose Vars, if a ``rose-suite.conf`` file is present.
```

File: cylc/flow/scripts/install.py
Comment (datamel) [inline cylc/flow/scripts/install.py:148]:
```suggestion
                "`rose-suite.conf` file. "
```

**Review by datamel — COMMENTED** [review]
I have worked my way through the bulk of this PR. 
Couple of comments. 
All tests pass for me locally. I have also read through code changes.
Will re-review when the 
`TODO: restore CYLC_SUITE_* job environment variables for back-compat` has been completed.
This change will break ui so will wait on giving explicit approval before a companion pr for cylc-uiserver is also approved.

**Comment by hjoliver** [comment]
> I have worked my way through the bulk of this PR.

Thanks @datamel, I'm sure that was mildly painful!

File: tests/unit/parsec/test_include.py
Comment (hjoliver) [reply tests/unit/parsec/test_include.py:19]: Good spotting. I hadn't thought about *other* meanings of the word "suite" :rofl:

Comment (hjoliver) [reply tests/unit/parsec/test_include.py:19]: I'll change this one back to suite.

File: cylc/flow/scripts/reinstall.py
Comment (hjoliver) [reply cylc/flow/scripts/reinstall.py:80]: (corrected several other instances of this too)

Comment (hjoliver) [reply cylc/flow/scripts/reinstall.py:80]: (which I see you've flagged below)

File: tests/functional/deprecations/01-cylc8-basic/validation.stderr
Comment (hjoliver) [reply tests/functional/deprecations/01-cylc8-basic/validation.stderr]: Yes, fixed.

**Comment by hjoliver** [comment]
Should be good to go now. No need to merge for beta-1 release though.

**Review by datamel — COMMENTED** [review]
I have reviewed the final files that I didn't get to yesterday... 
image!!!!
A couple of points....
* (Minor) On occasion you have replaced `suite server program` with `scheduler`, did you want to sed workflow server program and change them all to scheduler for consistency?
* (Minor) `etc/bin/run-functional-tests` has a couple of workflows that should be suites.
* (Possibly more major) I have checkout out sibling PRs and get errors when my run directory contains a contact file with old fields e.g CYLC_SUITE_UUID.. this returns traceback to the user
image
Maybe we want some back compatibility, since users may encounter this problem?

Compatibility for cylc review was also raised by @dpmatthews, @wxtim has added this to his to do list.

**Comment by hjoliver** [comment]
Good points @datamel, I'll address them soon...

File: cylc/flow/cfgspec/workflow.py
Comment (MetRonnie) [inline cylc/flow/cfgspec/workflow.py:1512]: I have a feeling this is not needed, I thought
```python
    u.deprecate(
        '8.0.0',
        ['runtime', '__MANY__', 'suite state polling'],
        ['runtime', '__MANY__', 'workflow state polling'],
    )
```
above would take care of it.

File: cylc/flow/cfgspec/workflow.py
Comment (MetRonnie) [inline cylc/flow/cfgspec/workflow.py:1497]: This ought to have be tested in `tests/functional/deprecations/01-cylc8-basic.t`

File: cylc/flow/cfgspec/workflow.py
Comment (MetRonnie) [inline cylc/flow/cfgspec/workflow.py:1518]: This ought to have be tested in `tests/functional/deprecations/01-cylc8-basic.t`

File: cylc/flow/etc/job.sh
Comment (MetRonnie) [inline cylc/flow/etc/job.sh:109]: If these are deprecated, can't the left-hand side be left as is (and the corresponding entries below removed)?

File: cylc/flow/etc/syntax/cylc-mode.el
Comment (MetRonnie) [inline cylc/flow/etc/syntax/cylc-mode.el:117]:
```suggestion
(add-to-list 'auto-mode-alist '("suite*.rc" . cylc-mode))
```

File: cylc/flow/etc/syntax/cylc-mode.el
Comment (MetRonnie) [inline cylc/flow/etc/syntax/cylc-mode.el:120]:
```suggestion
    '(("suite*.rc" . cylc-mode)
```

File: cylc/flow/etc/syntax/cylc.lang
Comment (MetRonnie) [inline cylc/flow/etc/syntax/cylc.lang:44]:
```suggestion
    <property name="globs">suite*.rc;*.cylc</property>
```

File: cylc/flow/etc/syntax/cylc.vim
Comment (MetRonnie) [inline cylc/flow/etc/syntax/cylc.vim:14]:
```suggestion
"|  au! BufRead,BufnewFile *suite*.rc   set filetype=cylc
```

File: cylc/flow/etc/syntax/cylc.xml
Comment (MetRonnie) [inline cylc/flow/etc/syntax/cylc.xml:3]:
```suggestion
<language name="Cylc flow.cylc" section="Configuration" extensions="suite*.rc;*.cylc" mimetype="" version="1.0" kateversion="2.0" author="" license="GPL">
```

File: cylc/flow/exceptions.py
Comment (MetRonnie) [inline cylc/flow/exceptions.py:94]: `ServiceFileError` would be less lengthy, but I appreciate it might be a pain to change again

File: cylc/flow/scripts/cylc.py
Comment (MetRonnie) [inline cylc/flow/scripts/cylc.py:140]:
```suggestion
    'get-suite-config':
        'cylc get-suite-config has been replaced by cylc config',
```

Also I think an alias or dead end for `suite-state` (now `workflow-state`) should be added

File: cylc/flow/cfgspec/workflow.py
Comment (hjoliver) [reply cylc/flow/cfgspec/workflow.py:1512]: Yes you're right - thought I had tested that to check, but evidently not!

File: cylc/flow/etc/job.sh
Comment (hjoliver) [reply cylc/flow/etc/job.sh:109]: Oops, yeah, looks like a `sed` fail (user error).

File: cylc/flow/cfgspec/workflow.py
Comment (hjoliver) [reply cylc/flow/cfgspec/workflow.py:1497]: Done.

File: cylc/flow/cfgspec/workflow.py
Comment (hjoliver) [reply cylc/flow/cfgspec/workflow.py:1518]: Turns out this item was already obsoleted, so I've removed the deprecation, and added the obsolete item to the test.

File: cylc/flow/exceptions.py
Comment (hjoliver) [reply cylc/flow/exceptions.py:94]: Easy to do with a `perl` or `sed` one-liner.

File: cylc/flow/scripts/cylc.py
Comment (hjoliver) [reply cylc/flow/scripts/cylc.py:140]: Done.

**Comment by hjoliver** [comment]
(closed and reopened to trigger Actions)

**Comment by hjoliver** [comment]
@datamel - 

> (Possibly more major) I have checkout out sibling PRs and get errors when my run directory contains a contact file with old fields e.g CYLC_SUITE_UUID.. this returns traceback to the user

Last commit fixes this. (It fixes `cylc scan` here, in case of old contact files, and the hub-reported errors if you `pip install` this branch into your ui-server).

**Comment by hjoliver** [comment]
I think the test failures are spurious - something weird just happened (see comment on Element) ... but I'll come back to check later.

**Comment by hjoliver** [comment]
Tests passing 🎉

**Comment by hjoliver** [comment]
Added extended back-compat comment as suggested.  With `[skip ci]` as it was just a comment.

File: cylc/flow/cfgspec/workflow.py
Comment (MetRonnie) [inline cylc/flow/cfgspec/workflow.py:238]: These two appear to have snuck in as a regression during #3963, and should be removed. https://github.com/cylc/cylc-flow/issues/3696#issuecomment-713802640 shows that they are meant to be obsolete

**Review by MetRonnie — CHANGES_REQUESTED** [review]

File: cylc/flow/cfgspec/workflow.py
Comment (hjoliver) [reply cylc/flow/cfgspec/workflow.py:238]: Good spot, I was actually wondering about that :+1:

File: tests/functional/deprecations/03-workflow.t
Comment (hjoliver) [reply tests/functional/deprecations/03-workflow.t:4]: Fair call.

File: cylc/flow/cfgspec/workflow.py
Comment (MetRonnie) [reply cylc/flow/cfgspec/workflow.py]: I got quite confused for a while 🤪

Comment (MetRonnie) [reply cylc/flow/cfgspec/workflow.py]: The `[environment]` section should be removed too, but can leave to another PR

File: tests/functional/restart/54-incompatible-db/db.sqlite3
Comment (MetRonnie) [inline tests/functional/restart/54-incompatible-db/db.sqlite3:8]: This file should be reverted, as it's recreating a Cylc 7 DB

**Review by MetRonnie — CHANGES_REQUESTED** [review]

File: cylc/flow/rundb.py
Comment (MetRonnie) [inline cylc/flow/rundb.py:170]: Just realised, the changes to the DB table names will mean anything last run on 8.0b1 or below won't restart.
```suggestion
    RESTART_INCOMPAT_VERSION = "8.0b1"  # Can't restart if <= this version
```

**Review by MetRonnie — CHANGES_REQUESTED** [review]

**Comment by MetRonnie** [comment]
A binary file `tests/integration/.test_client.py.swo` has appeared, I'm guessing by mistake?

**Review by MetRonnie — CHANGES_REQUESTED** [review]
Finally finished reviewing all the files 😅

File: cylc/flow/scheduler.py
Comment (hjoliver) [reply cylc/flow/scheduler.py:1637]: Huh, that one must have crept back in by rebase.

File: cylc/flow/rundb.py
Comment (hjoliver) [reply cylc/flow/rundb.py:170]: Yep, good spot.

File: cylc/flow/cfgspec/workflow.py
Comment (hjoliver) [reply cylc/flow/cfgspec/workflow.py]: Done.

File: tests/functional/restart/54-incompatible-db/db.sqlite3
Comment (hjoliver) [reply tests/functional/restart/54-incompatible-db/db.sqlite3:8]: The simple reversion works fine, it still causes the same error on attempting a restart.

**Comment by hjoliver** [comment]
> Finally finished reviewing all the files :sweat_smile: 

Thanks, excellent reviewing! :bouquet:

**Review by MetRonnie — APPROVED** [review]

**Review by datamel — APPROVED** [review]
I can confirm the issue that resulted in traceback on ui-server has been fixed.
Approved.

**Comment by oliver-sanders** [comment]
> thank @@#$$@# for easy command-line automation

Couldn't find any words in the English language which match that pattern, gonna have to be more specific.

File: cylc/flow/task_events_mgr.py
Comment (oliver-sanders) [inline cylc/flow/task_events_mgr.py:117]: This is a user-facing interface.

Should put `Suite` and `SuiteUUID` back in as deprecated fields so that event handlers can support Cylc7/8.

File: cylc/flow/task_events_mgr.py
Comment (oliver-sanders) [inline cylc/flow/task_events_mgr.py:1142]: Should keep `Suite` and `SuiteUUID` as deprecated fields (see above).

File: cylc/flow/workflow_events.py
Comment (oliver-sanders) [inline cylc/flow/workflow_events.py:109]: I think we want to keep `suite` as a deprecate field here too.

**Review by oliver-sanders — COMMENTED** [review]
Well that was fun!

I think we need to keep the "suite" fields we provide to ~suite~workflow/task event scripts (unless there's some clever logic handling that I didn't spot).

File: cylc/flow/task_events_mgr.py
Comment (hjoliver) [reply cylc/flow/task_events_mgr.py:117]: Good call. I've also done this for workflow (as opposed to task) events.

File: cylc/flow/workflow_events.py
Comment (hjoliver) [reply cylc/flow/workflow_events.py:109]: Yep. I've also handled bad mail footer templates here with a warning, instead of falling over dead.

**Review by oliver-sanders — APPROVED** [review]
:+1:

**Comment by oliver-sanders** [comment]
Brace for conflicts, it's going in!

**Pull request merged**
